controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1852,6 +1852,53 @@ int bml_internal::register_nw_map_update_cb(BML_NW_MAP_QUERY_CB pCB)
     return (BML_RET_OK);
 }
 
+int bml_internal::device_oper_radios_query(BML_DEVICE_DATA *device_data)
+{
+    // Initialize the promise for receiving the response
+    beerocks::promise<bool> prmDeviceDataGet;
+    m_prmDeviceDataGet = &prmDeviceDataGet;
+    int iOpTimeout     = RESPONSE_TIMEOUT; // Default timeout
+
+    beerocks_message::sDeviceData DeviceData;
+    m_device_data         = &DeviceData;
+    m_device_data->al_mac = network_utils::mac_from_string(device_data->al_mac);
+
+    register_topology_discovery_query();
```

Comment:
The function name is not OK - it should be `register_topology_discovery_response()`, you don't register to a "query".

Reply:
first i wrote it exactly like you propose but then i aligned to `register_nw_map_query_cb` naming.
I'll change it back.